add apm-server helm chart

pbecotte opened the pull request:
- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [x] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

Take a stab at adding the apm-server helm chart. Not sure about an open source image, or if there's a good way to flesh out the `goss` tests. I am NOT an apm-server expert, any suggestions are welcome. I have this running on my cluster currently.

elasticmachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

jmlrt commented:
related to https://github.com/elastic/helm-charts/issues/68

jmlrt commented:
Hi @pbecotte,
Thank you for this great PR!
Adding yourself Goss tests and Python tests is really nice 👍.
The first think I see is that `httpPort` is missing in your `values.yml`.
We'll do a full review soon.

jmlrt commented on apm-server/values.yaml:
```diff
@@ -0,0 +1,151 @@
+---
+# Comma separated list of hosts
```
`httpPort` is missing

pbecotte commented on apm-server/examples/default/test/goss.yaml:
```diff
@@ -0,0 +1,5 @@
+port:
+  tcp:8200:
```
Seems likely- but I don't really know that much about apm-server to write more useful tests... any ideas?

jmlrt replied:
For Logstash Helm Chart in #347 we decided to drop the `port` test and use a [http test](https://github.com/jmlrt/helm-charts/blob/master/logstash/examples/default/test/goss.yaml#L7-L19).

You could use this `http` test for APM (I don't know if there is a more relevant endpoint)

```
http:
  http://localhost:8200?pretty:
    status: 200
    timeout: 2000
    body:
      - '"version" : "7.4.0"'
```

pbecotte commented:
Addressed your comments- not sure how I had so many bugs in there- the tests passed at some point, but apparently I didn't run them before pushing? Not sure what would be a good test for apm server- maybe there is a health check url we should send a request to?

jmlrt commented:
jenkins test this please

jmlrt commented:
Hi @pbecotte,

Better late than never, we are thinking about merging this PR before the end of the month and release APM server chart in alpha for 7.6.0.

I think we still had a few comments in https://github.com/elastic/helm-charts/pull/324#pullrequestreview-315903104 and certainly a few others to come in the next days.

Are you still interested to work on it?

pbecotte commented:
Yeah, didn't want to get too far out in front of it, but would be happy to go through and update it.

jmlrt commented on apm-server/README.md:
```diff
@@ -0,0 +1,136 @@
+* The default APM Server configuration file for this chart is configured to use an 
+Elasticsearch endpoint as configured by the rest of these helm charts. This can
+easily be overridden in the config value apmConfig.apm-server.yml
+
```
When I mentionned a note, it was a bullet point like in other [README](https://github.com/elastic/helm-charts/blame/master/logstash/README.md#L15) (sorry for not being enough explicit).

jmlrt commented:
jenkins test this please

jmlrt commented:
jenkins test this please

jmlrt commented:
jenkins test this please
